src/modules/item/selectors.ts

```diff
@@ -164,18 +172,27 @@ export const getStatusByItemId = createSelector<
   EntityState['data'],
   Record<string, CollectionCuration>,
   Record<string, ItemCuration>,
+  Record<string, boolean>,
   Record<string, SyncStatus>
 >(
   getItems,
   getEntityByItemId,
   getCurationsByCollectionId,
   getItemCurationsByItemId,
-  (items, entitiesByItemId, curationsByCollectionId, itemCurationByItemId) => {
+  state => getMissingEntities(state),
+  (items, entitiesByItemId, curationsByCollectionId, itemCurationByItemId, missingEntities) => {
     const statusByItemId: Record<string, SyncStatus> = {}
     for (const item of items) {
+      const isMissingEntity = !!item.urn && missingEntities[item.urn]
+
       statusByItemId[item.id] = isThirdParty(item.urn)
         ? getStatusForTP(item, itemCurationByItemId[item.id])
-        : getStatusForStandard(item, item.collectionId ? curationsByCollectionId[item.collectionId] : null, entitiesByItemId[item.id])
+        : getStatusForStandard(
+            item,
+            item.collectionId ? curationsByCollectionId[item.collectionId] : null,
+            entitiesByItemId[item.id],
```

Comment:
If the item does not have an entity, `entitiesByItemId[item.id]` should be undefined, right? So if the item does not have an entity but is approved, there's an entity missing.

Reply:
you're right! When `entitiesByItemId[item.id]` is undefined, the item doesn't have an entity yet.

it could happen that the entities are being fetched or don't exist... 

added an extra validation to know when we're fetching the entities to properly know if it's missing or loading